dht-node/src/dht_node/index.test.ts

```diff
@@ -155,6 +162,135 @@ describe('federation', () => {
+          const result = await DbCommunity.find({ foreign: false })
+          expect(result).toEqual(
+            expect.arrayContaining([
+              expect.objectContaining({
+                id: expect.any(Number),
+                foreign: false,
+                url: CONFIG.FEDERATION_COMMUNITY_URL + '/api/',
+                publicKey: expect.any(Buffer),
+                communityUuid: expect.any(String),
+                authenticatedAt: null,
+                name: CONFIG.COMMUNITY_NAME,
+                description: CONFIG.COMMUNITY_DESCRIPTION,
+                creationDate: expect.any(Date),
+                createdAt: expect.any(Date),
+                updatedAt: null,
+              }),
+            ]),
+          )
+          const valUUID = validateUUID(
+            result[0].communityUuid != null ? result[0].communityUuid : '',
+          )
+          const verUUID = versionUUID(
+            result[0].communityUuid != null ? result[0].communityUuid : '',
+          )
+          expect(valUUID).toEqual(true)
+          expect(verUUID).toEqual(4)
+        })
+        it('update the one in table communities', async () => {
+          const resultBefore = await DbCommunity.find({ foreign: false })
+          expect(resultBefore).toHaveLength(1)
+          const modifiedCom = DbCommunity.create()
+          modifiedCom.communityUuid = resultBefore[0].communityUuid
+          modifiedCom.creationDate = resultBefore[0].creationDate
+          modifiedCom.description = 'updated description'
+          modifiedCom.foreign = resultBefore[0].foreign
+          modifiedCom.id = resultBefore[0].id
+          modifiedCom.name = 'update name'
+          modifiedCom.publicKey = Buffer.from(
+            '1234567891abcdef7892abcdef7893abcdef7894abcdef7895abcdef7896abcd',
+          )
+          modifiedCom.url = 'updated url'
+          await DbCommunity.update(modifiedCom, { id: resultBefore[0].id })
+
+          await writeHomeCommunityEntry(modifiedCom.publicKey.toString())
+          const resultAfter = await DbCommunity.find({ foreign: false })
+          expect(resultAfter).toHaveLength(1)
+          expect(resultAfter).toEqual(
+            expect.arrayContaining([
+              expect.objectContaining({
+                id: resultBefore[0].id,
+                foreign: false,
+                url: CONFIG.FEDERATION_COMMUNITY_URL + '/api/',
+                publicKey: modifiedCom.publicKey,
+                communityUuid: resultBefore[0].communityUuid,
+                authenticatedAt: null,
+                name: CONFIG.COMMUNITY_NAME,
+                description: CONFIG.COMMUNITY_DESCRIPTION,
+                creationDate: expect.any(Date),
+                createdAt: expect.any(Date),
+                updatedAt: expect.any(Date),
+              }),
+            ]),
+          )
+        })
+      })
+
+      // skipped because ot timing problems in testframework
+      describe.skip('federated home community', () => {
```

Comment:
I have not tested this yet - I assume there ain't much to do for me anyways if @Mogge could not find the error - so please do not be blocked by me wanting to understand the problem here.